numpy/random/mtrand.pyx

```diff
@@ -3609,7 +3609,7 @@ cdef class RandomState:
 
     # Multivariate distributions:
     def multivariate_normal(self, mean, cov, size=None, check_valid='warn',
-                            tol=1e-8):
+                            mode="default", factor=None, tol=1e-8):
```

Comment:
IMO the API only needs one new argument, mode (or similar), Options are 'default', 'legacy', 'fast', and 'factor', so that `cov` is treated as a factor when mode=factor. 

Docstring should have an example with `mode='factor'` too which will help explain why this is here, and how it could be useful (speeds up repeated calls with same covariance or with same correaltion but different variances).

Reply:
I see what you mean and it definitely makes more sense to have one argument. How do you suggest I test for validity of the input factor when `mode='factor'` is used? Currently I compare `factor @ factor.T` to `cov` to test if its a valid input. So if cov is to be replaced by factor in such a case then some other check needs to be performed. Or is it fine to just assume the use has the correct input?

Reply:
@zoj613 I don't see an alternative to assuming the factor is correct, but you can check if it is square. The product will always be positive semidefinite so no worry there.

Reply:
Note that we add new arguments at the end in case someone is passing the current arguments by position. It would be nice if at some point we could transition to keyword-only, but that will be a long term project.